libs/src/services/creatorNode/index.js

```diff
@@ -111,13 +114,23 @@ class CreatorNode {
    * @param {object} metadata the creator metadata
    */
   async uploadCreatorContent (metadata) {
-    return this._makeRequest({
-      url: '/audius_users/metadata',
-      method: 'post',
-      data: {
-        metadata
-      }
-    })
+    // this does the actual validation before sending to the creator node
+    // if validation fails, validate() will throw an error
+    try {
+      this.schemas[SchemaValidator.userSchemaType].validate(metadata)
+      console.log('user metadata validation passed', metadata)
+
+      return this._makeRequest({
+        url: '/audius_users/metadata',
+        method: 'post',
+        data: {
+          metadata
+        }
+      })
+    } catch (e) {
+      console.error(e)
+      throw new Error(e)
```

Comment:
if an error is thrown here, will the creator node stop working? one of the parent methods is `addCreator()` and the line where this is used has no error handling:
```
const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(
      newMetadata
    )
```

perhaps instead of throwing an error, return `null` and handle as necessary?

Reply:
i realized throwing errors actually won't work because it can sometimes cause the dapp to crash so i changed all of these to console.error